futures-util/src/io/take.rs

```diff
@@ -0,0 +1,35 @@
+use futures_core::task::{Context, Poll};
+use futures_io::AsyncRead;
+use std::io;
+use std::pin::Pin;
+
+/// Future for the [`take`](super::AsyncReadExt::take) method.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Take<R: Unpin> {
+    inner: R,
+    limit: u64,
+}
+
```

Comment:
Does this need a manual `Unpin` impl like some of the other adapters?

Reply:
As u64 [always implements Unpin](https://doc.rust-lang.org/nightly/std/primitive.u64.html#impl-Unpin), manual Unpin implementation should not be necessary here, but it can also be added for clarity:
```rust
impl<R: Unpin> Unpin for Take<R> {}
```